unit_test/main_options_test.py

```diff
@@ -0,0 +1,202 @@
+
+    expected_repair = repair_command or get_default_repair_command(platform)
+    assert intercepted_build_args.kwargs['repair_command'] == expected_repair
+
+
+@pytest.mark.parametrize('environment', [
+    {},
+    {'something': 'value'},
+    {'something': 'value', 'something_else': 'other_value'}
+])
+@pytest.mark.parametrize('platform_specific', [False, True])
+def test_environment(environment, platform_specific, platform, intercepted_build_args, monkeypatch):
+    env_string = ' '.join(['{}={}'.format(k, v) for k, v in environment.items()])
+    if platform_specific:
+        monkeypatch.setenv('CIBW_ENVIRONMENT_' + platform.upper(), env_string)
+        monkeypatch.setenv('CIBW_ENVIRONMENT', 'overwritten')
+    else:
+        monkeypatch.setenv('CIBW_ENVIRONMENT', env_string)
+
+    main()
+
+    intercepted_environment = intercepted_build_args.kwargs['environment']
+    assert isinstance(intercepted_environment, ParsedEnvironment)
+    assert intercepted_environment.as_dictionary({}) == environment
+
+
+@pytest.mark.parametrize('test_requires', [None, 'requirement other_requirement'])
+@pytest.mark.parametrize('platform_specific', [False, True])
+def test_test_requires(test_requires, platform_specific, platform, intercepted_build_args, monkeypatch):
+    if test_requires is not None:
+        if platform_specific:
+            monkeypatch.setenv('CIBW_TEST_REQUIRES_' + platform.upper(), test_requires)
+            monkeypatch.setenv('CIBW_TEST_REQUIRES', 'overwritten')
+        else:
+            monkeypatch.setenv('CIBW_TEST_REQUIRES', test_requires)
+    
+    main()
+    
+    assert intercepted_build_args.kwargs['test_requires'] == (test_requires or '').split()
+
+
+@pytest.mark.parametrize('test_extras', [None, 'extras'])
+@pytest.mark.parametrize('platform_specific', [False, True])
+def test_test_extras(test_extras, platform_specific, platform, intercepted_build_args, monkeypatch):
+    if test_extras is not None:
+        if platform_specific:
+            monkeypatch.setenv('CIBW_TEST_EXTRAS_' + platform.upper(), test_extras)
+            monkeypatch.setenv('CIBW_TEST_EXTRAS', 'overwritten')
+        else:
+            monkeypatch.setenv('CIBW_TEST_EXTRAS', test_extras)
+    
+    main()
+    
+    assert intercepted_build_args.kwargs['test_extras'] == ('[' + test_extras + ']' if test_extras else '')
+
+
+@pytest.mark.parametrize('test_command', [None, 'test --command'])
+@pytest.mark.parametrize('platform_specific', [False, True])
+def test_test_command(test_command, platform_specific, platform, intercepted_build_args, monkeypatch):
+    if test_command is not None:
+        if platform_specific:
+            monkeypatch.setenv('CIBW_TEST_COMMAND_' + platform.upper(), test_command)
+            monkeypatch.setenv('CIBW_TEST_COMMAND', 'overwritten')
+        else:
+            monkeypatch.setenv('CIBW_TEST_COMMAND', test_command)
+
+    main()
+
+    assert intercepted_build_args.kwargs['test_command'] == test_command or ''
```

Comment:
This condition might be wonky, also, I think it's equivalent to 

    assert (intercepted_build_args.kwargs['test_command'] == test_command) or ''

which is equivalent to:

    assert intercepted_build_args.kwargs['test_command'] == test_command

the logical operators e.g. `and` `or` are evaluated after the comparison ones... e.g. 

```python
>>> if 'asd' == 1 or 2:
...    print('wat')
... 
wat
```

Maybe this should be 
```python
    assert intercepted_build_args.kwargs['test_command'] == (test_command or '')
```

Reply:
OK, yes, indeed. Weird, I don't know how that from my head onto the screen! Even more so because `test_command` will be `None` if not set, rather than `''`. Thanks for catching that!